packages/rancher-istio/charts/templates/_helpers.tpl

```diff
@@ -10,3 +10,23 @@
 {{- "" -}}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "tracing.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "tracing.fullname" -}}
+{{- if .Values.tracing.fullnameOverride -}}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- $name := default "rancher-tracing" .Values.nameOverride -}}
```

Comment:
Hmm since this is all part of the `rancher-istio` chart, should there be a separate `rancher-tracing` naming convention? Seems like it should be packaged as a separate chart (similar to rancher-pushprox) if we're going with that convention.

Reply:
were there any other changes you made to stuff that you pulled from upstream we need to take a closer look at?

Reply:
Yeah, I was kind of going back and forth on this, so definitely open to changes.

 We are not pulling from an upstream chart, so I would be making a chart similar to how I made the istio chart. For this reason it made sense to just bundle it in with istio as just templates or as a chart dependency. Since its only a few templates, I opted to not make it a nested chart inside the istio chart. I could see why it might be better to do this so that is does allow for that separate naming conventions. Do you think this would be a better approach, or do you feel like we should still go in the direction of rancher-pushprox. Either one would be simple to change to.

Essentially the changes I added are a mixture of our Istio V1 jaeger integration https://github.com/rancher/system-charts/tree/dev-v2.5/charts/rancher-istio/1.5.900/charts/tracing, and the Istio quick start example https://istio.io/latest/docs/ops/integrations/jaeger/#option-1-quick-start and there wasn't any significant deviation from these

Reply:
Discussed this offline, I think it makes sense for `rancher-tracing` to be a separate chart because:
1. it can / should be deployable without requiring rancher-istio
2. rancher-tracing is not production-grade, but rancher-istio is
3. we want to use two separate naming schemes, so a separate chart makes sense instead of one chart creating multiple resources with different naming schemes (e.g. `rancher-istio-*` v.s. `rancher-tracing-*`)

Since they are mostly independent, we should go with the new-chart-pulled-in-as-a-subchart approach rather than adding it into our existing chart.

Reply:
We can separate it out, but it will still live as a sub-chart of Istio. For now it will be hidden so it can't be installed separately, and since its a non-production service it probably shouldn't be installable separately.